---
title: "QUESTION11"
author: "Trevor Schaff"
date: "November 5, 2015"
output: html_document
---

```{r reading in data}
data_exon <- read.csv("data/Homo_sapiens.GRCh38.82.abinitio_exons.gtf", sep="\t")
data_transcript <- read.csv("data/Homo_sapiens.GRCh38.82.abinitio_transcripts.gtf", sep="\t")
names(data_exon)[1]<-"chromosome"
names(data_exon)[4]<-"start"
names(data_exon)[5]<-"end"

names(data_transcript)[1]<-"chromosome"
names(data_transcript)[4]<-"start"
names(data_transcript)[5]<-"end"

```

```{r making new length column}
data_exon$length <- (data_exon$end - data_exon$start)
data_transcript$length <- (data_transcript$end - data_transcript$start)

data_exon_clean <- data_exon[data_exon$chromosome == "1" | data_exon$chromosome == "2"| data_exon$chromosome == "3" | data_exon$chromosome == "4" | data_exon$chromosome == "5" | data_exon$chromosome == "6" | data_exon$chromosome == "7" | data_exon$chromosome == "8" | data_exon$chromosome == "9" | data_exon$chromosome == "10" | data_exon$chromosome == "11" | data_exon$chromosome == "12" | data_exon$chromosome == "13" | data_exon$chromosome == "14" | data_exon$chromosome == "15" | data_exon$chromosome == "16" | data_exon$chromosome == "17" | data_exon$chromosome == "18" | data_exon$chromosome == "19" | data_exon$chromosome == "20" | data_exon$chromosome == "21" | data_exon$chromosome == "22" | data_exon$chromosome == "X" | data_exon$chromosome == "Y",]

data_transcript_clean <- data_transcript[data_transcript$chromosome == "1" | data_transcript$chromosome == "2"| data_transcript$chromosome == "3" | data_transcript$chromosome == "4" | data_transcript$chromosome == "5" | data_transcript$chromosome == "6" | data_transcript$chromosome == "7" | data_transcript$chromosome == "8" | data_transcript$chromosome == "9" | data_transcript$chromosome == "10" | data_transcript$chromosome == "11" | data_transcript$chromosome == "12" | data_transcript$chromosome == "13" | data_transcript$chromosome == "14" | data_transcript$chromosome == "15" | data_transcript$chromosome == "16" | data_transcript$chromosome == "17" | data_transcript$chromosome == "18" | data_transcript$chromosome == "19" | data_transcript$chromosome == "20" | data_transcript$chromosome == "21" | data_transcript$chromosome == "22" | data_transcript$chromosome == "X" | data_transcript$chromosome == "Y",]

#install.packages("plyr")
library(plyr)
std <- function(x) sd(x)/sqrt(length(x))
exon_data <- ddply(data_exon_clean,~chromosome,summarise,mean=format(round(mean(length),2)),sd=format(round(sd(length),2)),se=format(round(std(length),2)))
transcript_data <- ddply(data_transcript_clean,~chromosome,summarise,mean=format(round(mean(length),2)),sd=format(round(sd(length),2)),se=format(round(std(length),2)))

names(exon_data)[3]<-"sd_exon"
names(exon_data)[4]<-"se_exon"

names(transcript_data)[3]<-"sd_transcript"
names(transcript_data)[4]<-"se_transcript"

names(exon_data)[2]<-"exon_length"
names(transcript_data)[2]<-"transcript_length"

recom<-merge(exon_data, transcript_data, by.x="chromosome", by.y="chromosome")

write.table(recom, sep='\t')

```

```{r histograms}
#install.packages("ggplot2")
library(ggplot2)
x_chrom <- subset(data_exon_clean, chromosome == "X", c(chromosome, length))

y_chrom <- subset(data_exon_clean, chromosome == "Y", c(chromosome, length))

hist(x_chrom$length)
hist(y_chrom$length)
```

```{r chrom length vs exon length}
chrlengths<- matrix(c(248956422,242193529,198295559,190214555,181538259,170805979,159345973,145138636,138394717,133797422,135086622,133275309,114364328,107043718,101991189,90338345,83257441,80373285,58617616,64444167,46709983,50818468,156040895,57227415))

names(chrlengths)[1]<-"chrlengths"

chrcols <- matrix(c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,"X","Y"))

lengths_with_chrs<-transform(chrlengths, new.col = chrcols)
names(lengths_with_chrs)[1]<-"chr_length"
names(lengths_with_chrs)[2]<-"chromosome"
average_exon_lengths <-matrix(c(171.81,170.24,167.44,180.71,179.45,175.64,175.3,172.36,174.12,172.48,178.05,166.45,176.77,177.29,172.85,172.86,171.88,174.18,196.75,168.64,175.69,176.25,193.39,191.06))

final_dataset <-transform(lengths_with_chrs, new.col = average_exon_lengths)
names(final_dataset)[1]<-"chr_length"
names(final_dataset)[3]<-"mean_exon_length"

cor(final_dataset$chr_length, final_dataset$mean_exon_length)
plot(final_dataset$chr_length, final_dataset$mean_exon_length)


```



There is not a significant relationship between the length of a chromosome and its corresponding mean exon length. The correlation coefficient is -0.25.


